File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        MainObj={}
        
        for n in nums:
            if n in MainObj:
                MainObj[n]+=1
            else:
                MainObj[n]=1
            if MainObj[n]>len(nums)//2:
                return n
                


            
                
            
           

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        // Moores Vooting Algo:
        int n=nums.size();
        int ans=0, freq=0;
        for(int i=0;i<n;i++){
            if(freq==0){
                ans=nums[i];
            }
            if(ans==nums[i]){
                freq++;
            }else{
                freq--;
            }
        }
        return ans;
        
    }
};